### DISCUSSION TITLE
Actual performance of IVF_SQ8

### DISCUSSION THREAD

**Discussion by nilsertle** [milvus-io/milvus#24980] [Q&A and General discussion]
Dear Engineering Team of Milvus,

Great work! You have done a fantastic job, especially with your PyMilvus client. I am Nils Ertle and currently writing my thesis where I am trying to achieve optimal results for different applications with different needs (accuracy, performance, and memory utilization).

I have stumbled across one of your tests where you employed the IVF_SQ8 indexing algorithm. (https://milvus.io/docs/performance_faq.md) However, you also provide a performance test on your Zilliz blog (https://zilliz.com/blog/Accelerating-Similarity-Search-on-Really-Big-Data-with-Vector-Indexing?__hstc=175614333.5bfda607ac23d7ee41111285a56c2b1a.1684497817570.1686413012231.1686416464909.16&__hssc=175614333.6.1686416464909&__hsfp=4265922327).
<img width="899" alt="image" src="https://github.com/milvus-io/milvus/assets/89461027/eacf06af-14e3-4454-bfff-516f5fdb717e">
<img width="768" alt="image" src="https://github.com/milvus-io/milvus/assets/89461027/0748c151-b015-4634-808d-185e7b8de5bc">
I am a bit confused since these two results show quite different results. If you could let me know which dataset you used (how many entries) for the test provided on milvus docs? And if you could also tell me why the results of the zilliz blog are a lot worse then the ones on the milvus blog?

I also employed my own tests where I have more similar results to the tests of Zilliz. I tested on a tiny version of ImageNet (100,000 entries). My test results look like the following:
![TPQ_IVF_SQ8](https://github.com/milvus-io/milvus/assets/89461027/bd8ab591-cf09-42dd-a281-e38df5d331d8)
Here is clearly visible that a higher nprobe value results in a higher "time per query". By the way, if you also know how to get more stable results I would be very thankful for the feedback. I am using a ec2 instance and run the standalone docker compose file. I have 8 vCPUs and 32 GB of RAM.

I hope to hear from you soon. I am very interested in the answer. Keep up the great work.

Yours sincerly
Nils Ertle

**Comment by xiaofan-luan** [comment]
I think one of the graph shows latency and the other one show QPS? and the test dataset might be different also, one is 50M SIFT128 and the other one is 100M SIFT128.
But the overall the trend is there.  I think nlist=4000 should be optimized for 100k entires but your test seems to show nlist larger to be better. That might be something interested to explore on

**Reply by nilsertle** [reply]
Thank you for the replay. I think you are mistaken though. Neither my test nor the test of Zilliz are testing the QPS but TPQ (time per query). That's why for my example larger nlist result in worse results, just like the tests from Zilliz show.

What exactly do you mean by latency? In the docs it states that it is testing the query performance which looks pretty much like TPQ. Am I wrong here? 
<img width="874" alt="image" src="https://github.com/milvus-io/milvus/assets/89461027/7886c4b2-b8bf-47fa-b46d-1050b84df9f5">

**Reply by xiaofan-luan** [reply]
Under most of the time QPS will be the opposite trend of TPQ.  Less TPQ means more QPS
